### DISCUSSION TITLE
The address field is not able to distinguish ID and FQDN

### DISCUSSION THREAD

**Discussion by Livy-Inverse** [rustdesk/rustdesk#9573] [General]
### Bug Description

The main GUI of RustDesk has a single textbox field for entering the destination of the remote computer. By default it is the ID of the remote computer. By enabling the Settings/Security/Security/Enable direct IP access, we can also input an IP address or a Fully Qualified Domain Name (FQDN) of the remote computer to connect to.

The problem is the ability to set a custom ID in Settings/Security/ID, which allows a-zA-Z0-9_. When I try to enter the FQDN of the remote computer in the address field, RustDesk was confused and think that it is a text based ID and fails to connect.

### How to Reproduce

Enter the FQDN into the address field, e.g: **hostname.sub.domain.tld**

RustDesk ends up with an error which reads:
```
Connection error
ID does not exist
```
To workaround this issue, I have to add the port behind the FQDN, to make it realize that it is a FQDN instead of an ID, e.g: **hostname.sub.domain.tld:21118**

### Expected Behavior

RustDesk should be able to detect that it is an FQDN, not an ID, and connect successfully. There should be a GUI option for that. Relying on the dot character is not a good method, because the user may use Partially-Qualified Domain Name (PQDN) for convenience, e.g: only hostname without any dot.

### Operating system(s) on local side and remote side

Windows 11 -> Windows 10

### RustDesk Version(s) on local side and remote side

1.3.1 -> 1.3.1

### Screenshots

![image](https://github.com/user-attachments/assets/b5de71d3-ddce-40b4-9c4e-da38156da0da)

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/5833
